src/ghci/mod.rs

```diff
@@ -890,12 +931,36 @@ impl Ghci {
                 compilation_start.elapsed()
             );
         } else {
-            tracing::info!(
-                "{} Finished {} in {:.2?}",
-                "All good!".if_supports_color(Stdout, |text| text.green()),
-                event.event_noun(),
-                compilation_start.elapsed()
-            );
+            // Display any tracked warnings even if compilation succeeded
+            // but exclude files that were compiled in this cycle (to avoid duplicates)
+            if self.opts.track_warnings {
+                self.display_tracked_warnings_excluding_compiled(log).await;
+            }
+
+            let warning_count = if self.opts.track_warnings {
+                self.warning_count()
+            } else {
+                0
+            };
+
+            if warning_count > 0 {
+                tracing::info!(
+                    "{} Finished {} in {:.2?} ({} warning{} tracked)",
+                    "Compilation succeeded".if_supports_color(Stdout, |text| text.yellow()),
```

Comment:
One big design change here is that I purposefully picked a new phrase + color to signify "Compilation has finished successfully, but there are warnings present." Yellow + the succinct message here (distinct from the normal "All good!" is meant to signify that.